### DISCUSSION TITLE
Deleting a github classroom assignment gets a 406 error, as of 2024/08/13

### DISCUSSION THREAD

**Discussion by boyland** [community/community#135806] [GitHub Education] [answered]
### Hi everyone,

I'm trying to clean up after Spring semester, removing old github-classroom-created student repositories.  Starting yesterday (Tuesday 8/13) evening, trying to delete an assignment gets a 406 error.

Steps:
1. Click the trash icon
<img width="1176" alt="image" src="https://github.com/user-attachments/assets/1c1af06b-b4ca-4f8d-8e57-a306e4e16c0a">

2.  Indicate that I understand the problems:
<img width="473" alt="image" src="https://github.com/user-attachments/assets/bbdd607c-20c1-4398-99b9-3dd704a5ab85">

3.  Then type the name of the assignment
<img width="473" alt="image" src="https://github.com/user-attachments/assets/3c39a895-467a-40b2-989b-39644d16d7b5">

4. The following error pops up:
<img width="887" alt="image" src="https://github.com/user-attachments/assets/c8011467-1daf-4a14-afc3-b86767b566b7">

This only started happening yesterday evening.  I deleted several assignments earlier (waiting 10 minutes between each deletion because otherwise it doesn't finish deleting the repos) and then I started getting 406 every time.  I waited until the next day, in case some rate-limiting factor were happening, but it continued to fail today too.

**Comment by boyland** [comment]
Update: The same bad behavior is still happening as of today (2024/08/16)

**Comment by mwklose** [comment, marked as answer]
Coming from another instructor - I am noticing that the assignment copying/forking repo process is different this year. Last year, it would directly fork from the template repository, but now, it creates an intermediate template which it forks from. 

I got this error when I deleted this intermediate template (attempting to verify that my code/autograding worked somewhat as expected), and then tried to delete the assignment. 

The fix which just worked for me was to rename the assignment (I turned L00 into L00_deprecated, for instance), and then attempt deleting that.

**Reply by boyland** [reply]
Yes!  That worked -- in fact I don't need to change the assignment at all: just select "Edit" and then "Update Assignment" and then the assignment can be deleted.  (And it deleted the 100+ repos that had been created using it.)

**Comment by kingma-sbw** [comment]
For me renaming was not possible: "ErrorThat assignment could not be found"

**Comment by kingma-sbw** [comment]
Is someone at classroom.github.com looking at this?

**Reply by professordaehn** [reply]
It seems as if most questions go unanswered. This is very frustrating, not only for me, but for my students. This scenario begins with students not being able to accept the assignment in the first place. When I investigate, I find the assignment has been duplicated (all the details are the same except for the ID) and I am unable to delete either one of the duplicates. Some have suggested to merely update the assignment with no changes and that allowed them to delete the assignment. However, when I tried that, that too fails because it tries to update an assignment using the same prefix, which isn't allowed. I've literally had to change the name of the assignment and then after doing that I was able to delete the (renamed) assignment.
